token/src/presets/erc721/enumerable_mintable_burnable.cairo

```diff
@@ -0,0 +1,262 @@
+    impl ERC721MetadataImpl =
+        erc721_metadata_component::ERC721MetadataImpl<ContractState>;
+
+    #[abi(embed_v0)]
+    impl ERC721MetadataCamelImpl =
+        erc721_metadata_component::ERC721MetadataCamelImpl<ContractState>;
+
+    #[abi(embed_v0)]
+    impl ERC721OwnerImpl = erc721_owner_component::ERC721OwnerImpl<ContractState>;
+
+    impl ERC721ApprovalInternalImpl = erc721_approval_component::InternalImpl<ContractState>;
+    impl ERC721BalanceInternalImpl = erc721_balance_component::InternalImpl<ContractState>;
+    impl ERC721BurnableInternalImpl = erc721_burnable_component::InternalImpl<ContractState>;
+    impl ERC721EnumerableInternalImpl = erc721_enumerable_component::InternalImpl<ContractState>;
+    impl ERC721MetadataInternalImpl = erc721_metadata_component::InternalImpl<ContractState>;
+    impl ERC721MintableInternalImpl = erc721_mintable_component::InternalImpl<ContractState>;
+    impl ERC721OwnerInternalImpl = erc721_owner_component::InternalImpl<ContractState>;
+
+    mod Errors {
+        const INVALID_ACCOUNT: felt252 = 'ERC721: invalid account';
+        const UNAUTHORIZED: felt252 = 'ERC721: unauthorized caller';
+        const INVALID_RECEIVER: felt252 = 'ERC721: invalid receiver';
+        const WRONG_SENDER: felt252 = 'ERC721: wrong sender';
+        const SAFE_TRANSFER_FAILED: felt252 = 'ERC721: safe transfer failed';
+    }
+
+    #[storage]
+    struct Storage {
+        #[substorage(v0)]
+        erc721_approval: erc721_approval_component::Storage,
+        #[substorage(v0)]
+        erc721_balance: erc721_balance_component::Storage,
+        #[substorage(v0)]
+        erc721_burnable: erc721_burnable_component::Storage,
+        #[substorage(v0)]
+        erc721_enumerable: erc721_enumerable_component::Storage,
+        #[substorage(v0)]
+        erc721_metadata: erc721_metadata_component::Storage,
+        #[substorage(v0)]
+        erc721_mintable: erc721_mintable_component::Storage,
+        #[substorage(v0)]
+        erc721_owner: erc721_owner_component::Storage,
+    }
+
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    enum Event {
+        ERC721ApprovalEvent: erc721_approval_component::Event,
+        ERC721BalanceEvent: erc721_balance_component::Event,
+        ERC721BurnableEvent: erc721_burnable_component::Event,
+        ERC721EnumerableEvent: erc721_enumerable_component::Event,
+        ERC721MetadataEvent: erc721_metadata_component::Event,
+        ERC721MintableEvent: erc721_mintable_component::Event,
+        ERC721OwnerEvent: erc721_owner_component::Event,
+    }
+
+    #[abi(embed_v0)]
+    impl InitializerImpl of super::IERC721EnumInit<ContractState> {
+        fn initializer(
+            ref self: ContractState,
+            name: ByteArray,
+            symbol: ByteArray,
+            base_uri: ByteArray,
+            recipient: ContractAddress,
+            token_ids: Span<u256>
+        ) {
+            self.erc721_metadata.initialize(name, symbol, base_uri);
```

Comment:
we can add test on world owner & use initializable component same way than in other preset